Realm.Shared/Realm.cs

```diff
@@ -431,18 +431,36 @@ private RealmObject CreateObject(string className, out RealmObject.Metadata meta
             return result;
         }
 
-        internal RealmObject MakeObjectForRow(string className, RowHandle rowHandle)
+        internal RealmObject MakeObjectForRow(RealmObject.Metadata metadata, IntPtr rowPtr)
```

Comment:
I think this method can be refactored to call `MakeObjectForRow(metadata, CreateRowHandle(rowPtr, SharedRealmHandle))` instead.